trunk/modules/python/dataset/ingest/labeled.py

```diff
@@ -140,7 +140,13 @@ def hdf5Dataset(filepath, holdoutPercentage=.05, minTest=5,
     # Here we create the handles to the data buffers. This operates on the
     # assumption the dataset may not fit entirely in memory. The handles allow
     # data to overflow and ultimately be stored entirely on disk. 
-    imageShape = list(getImageDims(train[0][0], log))
+
+    # find the most common file size and use to get the image
+    # shape
+    commonSize = mostCommon(train, lambda t: os.path.getsize(t[0]))

```

Comment:
I'm wondering if this is overkill for our general case (ie most datasets are uniformly sized). Consider instead having a parameter that turns off these checks off, as it will increase our time dramatically. We could instead have try/except, if the logic to test this isn't too cumbersome.